# Подготовка файлов и программ

Некоторые программы ставятся дважды из разных источников для демонстрации различных вариантов в случае неполадок. Для оптимизации времени работы необходимо удалить лишний код.

In [29]:
!pip install macs2

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.0/2.0 MB 9.4 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
  Using cached Cython-0.29.37-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl.metadata (3.1 kB)
Using cached Cython-0.29.37-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (1.9 MB)
  Created wheel for macs2: filename=MACS2-2.2.9.1-cp310-cp310-linux_x86_64.whl size=1838776 sha256=cad4005f34e3f118fa910c48cb7b599250d70c196cb441f86b78955726dee56c
  Stored in directory: /root/.cache/pip/wheels/99/af/5d/88c35ca0afcbc95d9338d97ae48934f1e8e73ed9a8c9c98ae0
Successfully built macs2


In [1]:
!pip install -q condacolab

In [1]:
import condacolab

In [2]:
condacolab.install()

✨🍰✨ Everything looks OK!


In [3]:
!conda install -c bioconda seqtk

Channels:
 - bioconda
 - conda-forge
Platform: linux-64
Solving environment: \ | / - done


==> WARNING: A newer version of conda exists. <==
    current version: 23.11.0
    latest version: 24.1.2

Please update conda by running

    $ conda update -n base -c conda-forge conda



# All requested packages already installed.



In [4]:
!conda install -c bioconda trimmomatic

Channels:
 - bioconda
 - conda-forge
Platform: linux-64
Solving environment: | / - \ done


==> WARNING: A newer version of conda exists. <==
    current version: 23.11.0
    latest version: 24.1.2

Please update conda by running

    $ conda update -n base -c conda-forge conda



# All requested packages already installed.



In [42]:
import collections.abc
import collections
collections.Iterable = collections.abc.Iterable
!pip install intervene

In [6]:
!apt-get install bedtools

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
bedtools is already the newest version (2.30.0+dfsg-2ubuntu0.1).
0 upgraded, 0 newly installed, 0 to remove and 35 not upgraded.


In [7]:
!apt-get install bowtie2 samtools

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  libhts3 libhtscodecs2
Suggested packages:
  bowtie2-examples cwltool
The following NEW packages will be installed:
  bowtie2 libhts3 libhtscodecs2 samtools
0 upgraded, 4 newly installed, 0 to remove and 35 not upgraded.
Need to get 2,078 kB/2,131 kB of archives.
After this operation, 6,005 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhts3 amd64 1.13+ds-2build1 [390 kB]
Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 samtools amd64 1.13-4 [520 kB]
Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 bowtie2 amd64 2.4.4-1 [1,169 kB]
Fetched 2,078 kB in 1s (3,767 kB/s)
Selecting previously unselected package libhtscodecs2:amd64.
(Reading database ... 121802 files and directories currently installed.)
Preparing to unpack .../libhtscodecs2_1.1.1-3_amd64.deb ...
Un

In [8]:
!conda install -c bioconda bowtie2

Channels:
 - bioconda
 - conda-forge
Platform: linux-64
Solving environment: | / - \ done


==> WARNING: A newer version of conda exists. <==
    current version: 23.11.0
    latest version: 24.1.2

Please update conda by running

    $ conda update -n base -c conda-forge conda



## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - bowtie2


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    bowtie2-2.5.3              |  py310ha0a81b8_0        14.1 MB  bioconda
    libhwloc-2.9.3             |default_h554bfaf_1009         2.5 MB  conda-forge
    libxcrypt-4.4.36           |       hd590300_1          98 KB  conda-forge
    perl-5.32.1                | 7_hd590300_perl5        12.7 MB  conda-forge
    tbb-2021.11.0              |       h00ab1b0_1         191 KB  conda-forge
    ------------------------------------------------------------
             

## Установка FastQC

In [9]:
!java -version

openjdk version "21.0.2-internal" 2024-01-16
OpenJDK Runtime Environment (build 21.0.2-internal-adhoc.conda.src)
OpenJDK 64-Bit Server VM (build 21.0.2-internal-adhoc.conda.src, mixed mode, sharing)


In [10]:
!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip

--2024-03-03 10:35:09--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip
Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4
Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 10249221 (9.8M) [application/zip]
Saving to: ‘fastqc_v0.11.9.zip’

fastqc_v0.11.9.zip  100%[===================>]   9.77M  9.68MB/s    in 1.0s    

2024-03-03 10:35:11 (9.68 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]



In [11]:
!unzip fastqc_v0.11.9.zip

Archive:  fastqc_v0.11.9.zip
  inflating: FastQC/cisd-jhdf5.jar   
   creating: FastQC/Configuration/
  inflating: FastQC/Configuration/adapter_list.txt  
  inflating: FastQC/Configuration/contaminant_list.txt  
  inflating: FastQC/Configuration/limits.txt  
  inflating: FastQC/fastqc           
  inflating: FastQC/fastqc_icon.ico  
   creating: FastQC/Help/
   creating: FastQC/Help/1 Introduction/
   creating: FastQC/Help/1 Introduction/.svn/
  inflating: FastQC/Help/1 Introduction/.svn/entries  
   creating: FastQC/Help/1 Introduction/.svn/props/
   creating: FastQC/Help/1 Introduction/.svn/text-base/
  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  
   creating: FastQC/Help/1 Introduction/.svn/tmp/
   creating: FastQC/Help/1 Introduction/.svn/tmp/props/
  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  
   creating: FastQC/Help/2 Basic Operations/
   creating: FastQC/Help/2 Basic Operations/.svn/
  inflating: FastQC/Help/2 Basic

In [12]:
!chmod a+x FastQC/fastqc

In [13]:
!./FastQC/fastqc --help


            FastQC - A high throughput sequence QC analysis tool

SYNOPSIS

	fastqc seqfile1 seqfile2 .. seqfileN

    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] 
           [-c contaminant file] seqfile1 .. seqfileN

DESCRIPTION

    FastQC reads a set of sequence files and produces from each one a quality
    control report consisting of a number of different modules, each one of 
    which will help to identify a different potential type of problem in your
    data.
    
    If no files to process are specified on the command line then the program
    will start as an interactive graphical application.  If files are provided
    on the command line then the program will run with no user interaction
    required.  In this mode it is suitable for inclusion into a standardised
    analysis pipeline.
    
    The options for the program as as follows:
    
    -h --help       Print this help file and exit
    
    -v --version    Print the version of the program and exit

# Выравнивание

## Скачиваем чтения

Для примера чтения были взяты из эксперимента https://www.encodeproject.org/experiments/ENCSR000AKC/

In [14]:
# ChIP-seq на гистоновой метке
!wget https://www.encodeproject.org/files/ENCFF146YJK/@@download/ENCFF146YJK.fastq.gz
!wget https://www.encodeproject.org/files/ENCFF329HCY/@@download/ENCFF329HCY.fastq.gz
# ChIP-seq контроль
!wget https://www.encodeproject.org/files/ENCFF838NYU/@@download/ENCFF838NYU.fastq.gz

--2024-03-03 10:35:52--  https://www.encodeproject.org/files/ENCFF146YJK/@@download/ENCFF146YJK.fastq.gz
Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144
Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.
HTTP request sent, awaiting response... 307 Temporary Redirect
Location: https://encode-public.s3.amazonaws.com/2016/10/12/a863cc4e-8bb8-44dd-a214-0bf421926318/ENCFF146YJK.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF146YJK.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX5EHH4INP&Signature=rXHlnVbxpZd80f4n5TmXkqKWkBU%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEGMaCXVzLXdlc3QtMiJGMEQCIDg4SGSK%2FeWSaWuMm1YMWd7hK7bgN9ONsQBkR4stCRRlAiB4J9Wy7P6vb%2FBw1up%2BBD3GyY3uERk3pnZZW8l7YKv9ziqzBQhcEAAaDDIyMDc0ODcxNDg2MyIMLZxxa73z8wfNXO0yKpAFFJXqYcLqMuI4fZwSb4S1cC6clJfjRohGfZ%2Bgh6tBeh3Kh7h%2FNU0SEuLcrv96het7Y3aGpM4MUCIe%2B57upIm%2B0LBlFi4a0o5Vybgnru0uyrwmRSVUdGQmpXM%2ByMiUm1c3UxeueEh7f%2BvG3L8lCAnZkw4af8Ftvhjd4cePFMl1EllEi

In [15]:
# распаковка ахрива
!gzip -d /content/ENCFF146YJK.fastq.gz /content/ENCFF329HCY.fastq.gz /content/ENCFF838NYU.fastq.gz

## FastQC (или multiQC может быть использован в работе)

In [16]:
!./FastQC/fastqc /content/ENCFF146YJK.fastq
!./FastQC/fastqc /content/ENCFF329HCY.fastq
!./FastQC/fastqc /content/ENCFF838NYU.fastq

Started analysis of ENCFF146YJK.fastq
Approx 5% complete for ENCFF146YJK.fastq
Approx 10% complete for ENCFF146YJK.fastq
Approx 15% complete for ENCFF146YJK.fastq
Approx 20% complete for ENCFF146YJK.fastq
Approx 25% complete for ENCFF146YJK.fastq
Approx 30% complete for ENCFF146YJK.fastq
Approx 35% complete for ENCFF146YJK.fastq
Approx 40% complete for ENCFF146YJK.fastq
Approx 45% complete for ENCFF146YJK.fastq
Approx 50% complete for ENCFF146YJK.fastq
Approx 55% complete for ENCFF146YJK.fastq
Approx 60% complete for ENCFF146YJK.fastq
Approx 65% complete for ENCFF146YJK.fastq
Approx 70% complete for ENCFF146YJK.fastq
Approx 75% complete for ENCFF146YJK.fastq
Approx 80% complete for ENCFF146YJK.fastq
Approx 85% complete for ENCFF146YJK.fastq
Approx 90% complete for ENCFF146YJK.fastq
Approx 95% complete for ENCFF146YJK.fastq
Analysis complete for ENCFF146YJK.fastq
Started analysis of ENCFF329HCY.fastq
Approx 5% complete for ENCFF329HCY.fastq
Approx 10% complete for ENCFF329HCY.fastq
Appr

Далее необходимо проанализировать выдачу fastqc, произвести фильтрацию или подрезание чтений в случае необходимости. Для примера указано использование trimmomatic, также можно использовать seqtk.

In [17]:
!trimmomatic SE -phred33 ENCFF838NYU.fastq ENCFF838NYU_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \
  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36

TrimmomaticSE: Started with arguments:
 -phred33 ENCFF838NYU.fastq ENCFF838NYU_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
Automatically using 2 threads
java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:213)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:152)
	at org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)
	at org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)
	at org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)
	at org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)
	at org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)
	at org.usadellab.trimmomatic.TrimmomaticSE.run(Tri

In [18]:
!./FastQC/fastqc ENCFF838NYU_trimmed.fastq

Started analysis of ENCFF838NYU_trimmed.fastq
Approx 5% complete for ENCFF838NYU_trimmed.fastq
Approx 10% complete for ENCFF838NYU_trimmed.fastq
Approx 15% complete for ENCFF838NYU_trimmed.fastq
Approx 20% complete for ENCFF838NYU_trimmed.fastq
Approx 25% complete for ENCFF838NYU_trimmed.fastq
Approx 30% complete for ENCFF838NYU_trimmed.fastq
Approx 35% complete for ENCFF838NYU_trimmed.fastq
Approx 40% complete for ENCFF838NYU_trimmed.fastq
Approx 45% complete for ENCFF838NYU_trimmed.fastq
Approx 50% complete for ENCFF838NYU_trimmed.fastq
Approx 55% complete for ENCFF838NYU_trimmed.fastq
Approx 60% complete for ENCFF838NYU_trimmed.fastq
Approx 65% complete for ENCFF838NYU_trimmed.fastq
Approx 70% complete for ENCFF838NYU_trimmed.fastq
Approx 75% complete for ENCFF838NYU_trimmed.fastq
Approx 80% complete for ENCFF838NYU_trimmed.fastq
Approx 85% complete for ENCFF838NYU_trimmed.fastq
Approx 90% complete for ENCFF838NYU_trimmed.fastq
Approx 95% complete for ENCFF838NYU_trimmed.fastq
Analy

## Выравнивание на хромосому

Следует выбрать одну хромосому, потому что ресурсы google colab ограничены. В случае проблем, таких как Segmentation fault, error 139, следует выбрать хромосому меньшего размера.

Последовательность нуклеотидов хромосомы можно скачать по адресу https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/

In [19]:
!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz
!gzip -d chr14.fa.gz

--2024-03-03 10:52:42--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz
Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.198.53
Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.198.53|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 29295890 (28M) [application/x-gzip]
Saving to: ‘chr14.fa.gz’

chr14.fa.gz         100%[===================>]  27.94M  31.3MB/s    in 0.9s    

2024-03-03 10:52:43 (31.3 MB/s) - ‘chr14.fa.gz’ saved [29295890/29295890]



In [20]:
%%time
!bowtie2-build chr14.fa chromosome_index

Settings:
  Output files: "chromosome_index.*.bt2"
  Line rate: 6 (line is 64 bytes)
  Lines per side: 1 (side is 64 bytes)
  Offset rate: 4 (one in 16)
  FTable chars: 10
  Strings: unpacked
  Max bucket size: default
  Max bucket size, sqrt multiplier: default
  Max bucket size, len divisor: 4
  Difference-cover sample period: 1024
  Endianness: little
  Actual local endianness: little
  Sanity checking: disabled
  Assertions: disabled
  Random seed: 0
  Sizeofs: void*:8, int:4, long:8, size_t:8
Input files DNA, FASTA:
  chr14.fa
Building a SMALL index
Reading reference sizes
  Time reading reference sizes: 00:00:01
Calculating joined length
Writing header
Reserving space for joined string
Joining reference sequences
  Time to join reference sequences: 00:00:01
bmax according to bmaxDivN setting: 22642037
Using parameters --bmax 16981528 --dcv 1024
  Doing ahead-of-time memory usage test
  Passed!  Constructing with these parameters: --bmax 16981528 --dcv 1024
Constructing suffix-arr

In [21]:
!mkdir bowtie2_res

In [22]:
%%time
# Реплика 1
!bowtie2  -p 2 \
          -x chromosome_index \
          -U ENCFF146YJK.fastq \
          -S bowtie2_res/res_YJK.sam

15475541 reads; of these:
  15475541 (100.00%) were unpaired; of these:
    12660119 (81.81%) aligned 0 times
    564716 (3.65%) aligned exactly 1 time
    2250706 (14.54%) aligned >1 times
18.19% overall alignment rate
CPU times: user 7.51 s, sys: 1.02 s, total: 8.53 s
Wall time: 20min 57s


In [23]:
%%time
# Реплика 2
!bowtie2  -p 2 \
          -x chromosome_index \
          -U ENCFF329HCY.fastq \
          -S bowtie2_res/res_HCY.sam

20074587 reads; of these:
  20074587 (100.00%) were unpaired; of these:
    16553645 (82.46%) aligned 0 times
    684331 (3.41%) aligned exactly 1 time
    2836611 (14.13%) aligned >1 times
17.54% overall alignment rate
CPU times: user 5.78 s, sys: 767 ms, total: 6.54 s
Wall time: 16min 15s


In [24]:
%%time
# Контроль
!bowtie2  -p 2 \
          -x chromosome_index \
          -U ENCFF838NYU_trimmed.fastq \
          -S bowtie2_res/res_NYU.sam

11149442 reads; of these:
  11149442 (100.00%) were unpaired; of these:
    9143276 (82.01%) aligned 0 times
    471033 (4.22%) aligned exactly 1 time
    1535133 (13.77%) aligned >1 times
17.99% overall alignment rate
CPU times: user 3.28 s, sys: 485 ms, total: 3.77 s
Wall time: 9min 16s


Проанализируйте выдачу bowtie. Почему процент выравниваний получился именно таким?

Имеет смысл для дальнейшего анализа отобрать уникально картированные риды.

| |Всего ридов| Выровненных уникально | Выровненных неуникально | Невыровненных|   
|:-|:-:|:-:|:-:|:-:|
ENCFF146YJK | 15475541 | 564716 (3.65%) | 2250706 (14.54%) |  12660119 (81.81%) |
ENCFF329HCY | 20074587 | 684331 (3.41%) | 2836611 (14.13%) | 16553645 (82.46%) |
ENCFF838NYU | 11149442 | 471033 (4.22%) | 1535133 (13.77%) | 9143276 (82.01%)|

Преобразование .sam в .bam (не требуется если запускать macs2 на .sam файлах)

In [25]:
!samtools view -h -S -b -o bowtie2_res/res_YJK.bam bowtie2_res/res_YJK.sam
!samtools view -h -S -b -o bowtie2_res/res_HCY.bam bowtie2_res/res_HCY.sam
!samtools view -h -S -b -o bowtie2_res/res_NYU.bam bowtie2_res/res_NYU.sam

На этом моменте можно скачать себе на компьютер все полученные .bam или .sam файлы на случай сбоя.

## Peak calling

Помимо macs2 может быть использован пакет Homer для поиска пиков, а также macs3.

In [26]:
!mkdir macs2

In [33]:
!macs2 callpeak --broad -t bowtie2_res/res_YJK.sam \
    -c bowtie2_res/res_NYU.sam \
 	  -f SAM \
	  --outdir macs2 \
		-n ENCFF146YJK

INFO  @ Sun, 03 Mar 2024 14:58:40: 
# Command line: callpeak --broad -t bowtie2_res/res_YJK.sam -c bowtie2_res/res_NYU.sam -f SAM --outdir macs2 -n ENCFF146YJK
# ARGUMENTS LIST:
# name = ENCFF146YJK
# format = SAM
# ChIP-seq file = ['bowtie2_res/res_YJK.sam']
# control file = ['bowtie2_res/res_NYU.sam']
# effective genome size = 2.70e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff for narrow/strong regions = 5.00e-02
# qvalue cutoff for broad/weak regions = 1.00e-01
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 1000 bps and 10000 bps
# Broad region calling is on
# Paired-End mode is off
 
INFO  @ Sun, 03 Mar 2024 14:58:40: #1 read tag files... 
INFO  @ Sun, 03 Mar 2024 14:58:40: #1 read treatment tags... 
INFO  @ Sun, 03 Mar 2024 14:58:43:  3000000 
INFO  

In [35]:
!macs2 callpeak --broad -t bowtie2_res/res_HCY.sam \
    -c bowtie2_res/res_NYU.sam \
 	  -f SAM \
	  --outdir macs2 \
		-n ENCFF329HCY

INFO  @ Sun, 03 Mar 2024 15:00:07: 
# Command line: callpeak --broad -t bowtie2_res/res_HCY.sam -c bowtie2_res/res_NYU.sam -f SAM --outdir macs2 -n ENCFF329HCY
# ARGUMENTS LIST:
# name = ENCFF329HCY
# format = SAM
# ChIP-seq file = ['bowtie2_res/res_HCY.sam']
# control file = ['bowtie2_res/res_NYU.sam']
# effective genome size = 2.70e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff for narrow/strong regions = 5.00e-02
# qvalue cutoff for broad/weak regions = 1.00e-01
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 1000 bps and 10000 bps
# Broad region calling is on
# Paired-End mode is off
 
INFO  @ Sun, 03 Mar 2024 15:00:07: #1 read tag files... 
INFO  @ Sun, 03 Mar 2024 15:00:07: #1 read treatment tags... 
INFO  @ Sun, 03 Mar 2024 15:00:10:  4000000 
INFO  

In [32]:
!wget https://www.encodeproject.org/files/ENCFF319QES/@@download/ENCFF319QES.bed.gz
!gzip -d ENCFF319QES.bed.gz

--2024-03-03 14:55:14--  https://www.encodeproject.org/files/ENCFF319QES/@@download/ENCFF319QES.bed.gz
Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144
Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.
HTTP request sent, awaiting response... 307 Temporary Redirect
Location: https://encode-public.s3.amazonaws.com/2020/11/19/dcbfd699-78f0-48f2-b1dc-9744cb752d24/ENCFF319QES.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF319QES.bed.gz&AWSAccessKeyId=ASIATGZNGCNX5CUFVFVD&Signature=NWv0l98ZvT28Thw62J8PFFnXAAY%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEGcaCXVzLXdlc3QtMiJHMEUCIQDzgbuVk7KD9PWNyPLTYiTfKQ%2BltOHUZ9MbYqVMcYzE0gIgTIUZzHGo%2BvLvlOnk7RQRNvMphn%2FY3T99VCoNe5fGYdoqswUIYBAAGgwyMjA3NDg3MTQ4NjMiDK8gr7oqnvrNFaKhByqQBeYj%2FOgQ6Zb8c%2FzBCXI2fAX%2Fqd8%2B289mc6dzf58mBilhxCiARRxyb4pu%2BwMydlkxDtMjGeOprf3yCb2CH11Y1LeV5KSD104PF5kQgnFp0U2DHKKMAXwsT23I2YgmBUYhdS2a7l7dcxlRO62RJFkFD02AB7kTMjiipu%2FI8GxhMaRqgYa9QwJ

## Сравнение результатов

Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.

Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?

На данный момент версия пакета не работает с версией python установленной в colab по-умолчанию.

Пример смены версий и запуска пакета реализован в тетрадке по ссылке

https://colab.research.google.com/drive/1LbV-P0kibDrKtCfliYebrTq_FBAh80uu?usp=sharing

In [56]:
%env PYTHONPATH = # /env/python
#чистим переменную пути и ставим конду

!wget https://repo.anaconda.com/miniconda/Miniconda3-py38_4.12.0-Linux-x86_64.sh
!chmod +x Miniconda3-py38_4.12.0-Linux-x86_64.sh
!./Miniconda3-py38_4.12.0-Linux-x86_64.sh -b -f -p /usr/local
!conda update conda
#Then, append this path to install packages.

import sys
sys.path.append('/usr/local/lib/python3.8/site-packages')
#создаем среду
!conda create -n myenv python=3.6

env: PYTHONPATH=# /env/python
--2024-03-03 15:28:27--  https://repo.anaconda.com/miniconda/Miniconda3-py38_4.12.0-Linux-x86_64.sh
Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.130.3, 104.16.131.3, 2606:4700::6810:8203, ...
Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.130.3|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 76120962 (73M) [application/x-sh]
Saving to: ‘Miniconda3-py38_4.12.0-Linux-x86_64.sh’

Miniconda3-py38_4.1 100%[===================>]  72.59M   129MB/s    in 0.6s    

2024-03-03 15:28:28 (129 MB/s) - ‘Miniconda3-py38_4.12.0-Linux-x86_64.sh’ saved [76120962/76120962]

PREFIX=/usr/local
Unpacking payload ...
Solving environment: \ | / - \ WARNING conda.core.solve:_add_specs(610): pinned spec python=3.10 conflicts with explicit specs.  Overriding pinned spec.
| / - \ | / - \ | / - failed with initial frozen solve. Retrying with flexible solve.
Solving environment: | / - WARNING conda.c

In [57]:
%%shell
eval "$(conda shell.bash hook)"
conda activate myenv
pip install pybedtools
pip install intervene
pip install -q deeptools
conda install -c bioconda bedtools

     |████████████████████████████████| 12.5 MB 1.3 MB/s            
  Preparing metadata (setup.py) ... done
     |████████████████████████████████| 18.4 MB 310 kB/s             
     |████████████████████████████████| 14.8 MB 51.6 MB/s            
  Created wheel for pybedtools: filename=pybedtools-0.9.1-cp36-cp36m-linux_x86_64.whl size=14332521 sha256=5e5d54099b9d4fd4675afaacd69be362387b5ef8ec37d846d4c9f59f164bf1ff
  Stored in directory: /root/.cache/pip/wheels/56/c8/2d/e0e81f79dcdc1d164abb6b1192e6e5b33eaeb8bdcc1b15cb81
Successfully built pybedtools
     |████████████████████████████████| 5.5 MB 1.3 MB/s            
  Preparing metadata (setup.py) ... done
     |████████████████████████████████| 11.5 MB 46.1 MB/s            
     |████████████████████████████████| 9.5 MB 55.8 MB/s            
     |████████████████████████████████| 25.9 MB 1.2 MB/s             
     |████████████████████████████████| 292 kB 66.0 MB/s            
     |████████████████████████████████| 103 kB 66.1 MB

In [58]:
%%shell
eval "$(conda shell.bash hook)"
conda activate myenv
intervene --help

usage: intervene <subcommand> [options]

    Intervene: a tool for intersection and visualization of multiple genomic region and gene sets.
    For more details check documentation: http://intervene.readthedocs.io
    

positional arguments:
  {venn,upset,pairwise}
                        List of subcommands
    venn                Venn diagram of intersection of genomic regions or list sets (upto 6-way).
    upset               UpSet diagram of intersection of genomic regions or list sets.
    pairwise            Pairwise intersection and heatmap of N genomic region sets in <BED/GTF/GFF> format.

optional arguments:
  -h, --help            show this help message and exit
  -v, --version         show program's version number and exit
  -c, --cite            show citation information and exit


In [63]:
%%shell
eval "$(conda shell.bash hook)"
conda activate myenv
intervene venn -i macs2/ENCFF146YJK_peaks.broadPeak ENCFF319QES.bed --filenames --output venn_results/venn1.jpg


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ venn_results/venn1.jpg. 
Thank you for using Intervene!



In [61]:
%%shell
eval "$(conda shell.bash hook)"
conda activate myenv
intervene venn -i ENCFF319QES.bed macs2/ENCFF146YJK_peaks.broadPeak --filenames --output venn_results/venn2.jpg


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ venn_results/venn2.jpg. 
Thank you for using Intervene!



In [64]:
%%shell
eval "$(conda shell.bash hook)"
conda activate myenv
intervene venn -i macs2/ENCFF329HCY_peaks.broadPeak ENCFF319QES.bed --filenames --output venn_results/venn3.jpg


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ venn_results/venn3.jpg. 
Thank you for using Intervene!



In [66]:
%%shell
eval "$(conda shell.bash hook)"
conda activate myenv
intervene venn -i ENCFF319QES.bed macs2/ENCFF329HCY_peaks.broadPeak --filenames --output venn_results/venn4.jpg


Generating a 2-way "venn" diagram. Please wait...


Done! Please check your results @ venn_results/venn4.jpg. 
Thank you for using Intervene!

